controllers/numaresourcesoperator_controller.go

```diff
@@ -494,10 +506,10 @@ func (r *NUMAResourcesOperatorReconciler) syncNUMAResourcesOperatorResources(ctx
 		if err != nil {
 			return err
 		}
-		dssByMCP[mcpName] = nropv1.NamespacedName{
+		dsMCPPairs = append(dsMCPPairs, poolDaemonSet{mcpName, nropv1.NamespacedName{
 			Namespace: gdm.DaemonSet.GetNamespace(),
 			Name:      gdm.DaemonSet.GetName(),
-		}
+		}})
```

Comment:
for the future: let's add our clone of `client.ObjectKeyFromObject` (will become `NamespacedNameFromObject`)
in addition, please always use named initializers everywhere

Reply:
same as https://github.com/openshift-kni/numaresources-operator/pull/1052#discussion_r1808484818

Reply:
used metav.Object since we already call it in v1.